internal/proxy/proxy.go

```diff
@@ -784,7 +791,6 @@ func (c *Client) newSocketMount(ctx context.Context, conf *Config, pc *portConfi
 			version, err := c.dialer.EngineVersion(ctx, inst.Name)
 			if err != nil {
 				c.logger.Errorf("could not resolve version for %q: %v", inst.Name, err)
-				return nil, err
```

Comment:
We need to keep this line. Anything below it would fail if we didn't return early.

Reply:
The benefit of not exiting this function is that If the user specifies the instance without TCP port, it will allow the the listen call to assign a port so that the proxy can continue to run despite the EngineVersion failure. The question is, do we want this behaviour?

Reply:
I think we want to push users to specify a port globally (e.g., --port=5000) or on a per-instance basis.

The behavior we want to add in this PR, is to skip failed instanced for Unix sockets only. This way we still come up even if one of the instances is instance. But this behavior only applies for Unix sockets (which you've captured above).